### PR TITLE
TS-4355: Change assert condition for TS-3612

### PR DESCRIPTION
**Pull request by masaori335**
Originall, the condition of assert in HttpTunnel::main_handler(int event, void *data)
was same to the condition in HttpTunnel::get_consumer(VIO *vio) before TS-3612.
This commit make the condition in HttpTunnel::main_handler(int event, void *data)
same as condition in HttpTunnel::get_consumer(VIO *vio).

### CODE REVIEW

**Comment by masaori335** [comment]
@shinrich Could you take a look?

**Comment by shinrich** [comment]
Looks good to me.

**Comment by jpeach** [comment]
Which tests triggers this assertion? Could you summarize the explanation from [TS-4355](https://issues.apache.org/jira/browse/TS-4355) in the commit message?

**Comment by masaori335** [comment]
When I send HTTP/2 request to ATS (built with debug option), I always see this assertion is triggered.
It looks like `make check` and Regression Tests couldn't detect this. HTTP/2 tests in TSQA should detect this, but I didn't try them.

**Comment by masaori335** [comment]
Summary is added.
